### PR TITLE
Globals Styles Endpoint: Declare edit_css capability in links

### PR DESCRIPTION
**Pull request by Mamaduka**
Trac ticket: https://core.trac.wordpress.org/ticket/57526

Related Gutenberg PR: https://github.com/WordPress/gutenberg/pull/46815

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/rest-api/rest-global-styles-controller.php [hunk 1]
```diff
@@ -513,4 +513,23 @@ public function test_get_theme_items() {
...
+		$response = rest_do_request( $request );
+		$links    = $response->get_links();
+
+		// Admins can only edit css on single site.
+		if ( is_multisite() ) {
+			$this->assertArrayNotHasKey( 'https://api.w.org/action-edit-css', $links );
+		} else {
+			$this->assertArrayHasKey( 'https://api.w.org/action-edit-css', $links );
+		}
+	}
```
Comment (Mamaduka) [inline tests/phpunit/tests/rest-api/rest-global-styles-controller.php:534]: I used the post-controller tests as a blueprint here https://github.com/WordPress/wordpress-develop/blob/6.1/tests/phpunit/tests/rest-api/rest-posts-controller.php#L4903-L4918.

**Comment by Mamaduka** [comment]
cc @hellofromtonya

File: src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php [hunk 2]
```diff
@@ -442,6 +442,10 @@ protected function get_available_actions() {
 			$rels[] = 'https://api.w.org/action-publish';
 		}
 
+		if ( current_user_can( 'edit_css' ) ) {
+			$rels[] = 'https://api.w.org/action-edit-css';
+		}
```
Comment (hellofromtonya) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php:448]: There's an [unanswered question from the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/46815#discussion_r1058902276)

>I couldn't find documentation on how these URLs are generated/created. So I need to double-check if `https://api.w.org/action-edit-css` is the correct format.

Pinging @TimothyBJacobs @spacedmonkey for review.

Comment (Mamaduka) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php:448]: All similar action URLs lead to a 404 page - `https://api.wordpress.org/action-unfiltered-html`.

Comment (Mamaduka) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php:448]: @TimothyBJacobs, @spacedmonkey, can you confirm that it's okay to use this URL?

Comment (TimothyBJacobs) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php:448]: Sorry for missing this the first time round. The URLs don't actually point anywhere. That URL is fine to use.

Comment (Mamaduka) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php:448]: Thank you, @TimothyBJacobs!

**Comment by hellofromtonya** [comment]
Currently reviewing.

File: tests/phpunit/tests/rest-api/rest-global-styles-controller.php (hunk 1 above)
Comment (hellofromtonya) [reply tests/phpunit/tests/rest-api/rest-global-styles-controller.php:534]: Good job @Mamaduka 👏

**Review by hellofromtonya — APPROVED** [review]
* Confirmed this patch has the REST PHP code changes from the GB PR ✅ 
* Timothy confirmed the URL is okay ✅ 
* Tests included ✅ 

Ready for commit 👍

**Comment by hellofromtonya** [comment]
Committed via https://core.trac.wordpress.org/changeset/55177.

**Pull request closed without merging**